src/librustdoc/html/templates/type_layout.html

```diff
@@ -28,13 +28,15 @@ <h2 id="layout" class="section-header"> {# #}
         {% endfor %}
     </ul>
     {% endif %}
-    {# This kind of layout error can occur with valid code, e.g. if you try to
-    get the layout of a generic type such as `Vec<T>`. #}
+        {# This kind of layout error can occur with valid code.
+        It can happen for various reasons, e.g., if you try to get
+        the layout of an array with a length that is not a constant
+        and it has not a generic parameter such as
+        `[(); size_of::<<() as Trait>::Assoc>()]`. #}
         {% when Err(LayoutError::Unknown(_)) %}
         <p> {# #}
-            <strong>Note:</strong> Unable to compute type layout, {#+ #}
-            possibly due to this type having generic parameters. {#+ #}
-            Layout can only be computed for concrete, fully-instantiated types. {# #}
+            <strong>Note:</strong> Unable to compute type layout. {#+ #}
+            It can happen for various reasons. {# #}
```

Comment:
This sentence adds no value.

```suggestion
            <strong>Note:</strong> Unable to compute type layout. {# #}
```